@here/harp-mapview/lib/CameraUtils.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as THREE from "three";
+
+import { MAX_FOV_DEG, MIN_FOV_DEG } from "./FovCalculation";
+
+export namespace CameraUtils {
+    /**
+     * Computes a camera's vertical field of view for given focal length and viewport height.
+     * @beta
+     *
+     * @param focalLength - Focal length in pixels (see {@link computeFocalLength})
+     * @param height - Viewport height in pixels.
+     * @returns Vertical field of view in radians.
+     */
+    export function computeVerticalFov(focalLength: number, height: number): number {
+        return 2 * Math.atan(height / 2 / focalLength);
+    }
+
+    /**
+     * Computes a camera's horizontal field of view.
+     * @beta
+     *
+     * @param camera
+     * @returns Horizontal field of view in radians.
+     */
+    export function computeHorizontalFov(camera: THREE.PerspectiveCamera): number {
+        const vFov = THREE.MathUtils.degToRad(camera.fov);
+        return 2 * Math.atan(Math.tan(vFov / 2) * camera.aspect);
+    }
+
+    /**
+     * Set a camera's horizontal field of view.
+     * @internal
+     *
+     * @param camera
+     * @param hFov - The horizontal field of view in degrees.
+     */
+    export function setHorizontalFov(camera: THREE.PerspectiveCamera, hFov: number): void {
+        camera.fov = THREE.MathUtils.radToDeg(
+            2 * Math.atan(Math.tan(THREE.MathUtils.degToRad(hFov) / 2) / camera.aspect)
+        );
+    }
+
+    /**
+     * Sets a camera's vertical field of view.
+     * @internal
+     *
+     * @param camera
+     * @param fov - The vertical field of view in degress.
+     */
+    export function setVerticalFov(camera: THREE.PerspectiveCamera, fov: number): void {
+        camera.fov = THREE.MathUtils.clamp(fov, MIN_FOV_DEG, MAX_FOV_DEG);
+
+        let hFov = THREE.MathUtils.radToDeg(computeHorizontalFov(camera));
+
+        if (hFov > MAX_FOV_DEG || hFov < MIN_FOV_DEG) {
+            hFov = THREE.MathUtils.clamp(hFov, MIN_FOV_DEG, MAX_FOV_DEG);
+            setHorizontalFov(camera, hFov);
+        }
+    }
+
+    /**
+     * Computes a camera's focal length for a given viewport height.
+     * @beta
+     *
+     * @param vFov - Vertical field of view in degress.
```

Comment:
could this instead be in radians, to avoid the conversion

Reply:
done